Core/Src/uart_ring.c

```diff
@@ -0,0 +1,30 @@
+#include "uart_ring.h"
+#include "user_assert.h"
+
+static uint8_t temp;
+
+void UART_Ring_RxCallback(STM32F051_UART_t* pUART)
+{
+  RING_Append(&((UART_Ring_t*)pUART)->ring, (uint8_t)pUART->handle->RDR);
+  UART_ReceiveIT(pUART, &temp, 1);
+
+  ((UART_Ring_t*)pUART)->UART_RxCallback();
+}
+
+uint32_t UART_Ring_Init(UART_Ring_t* ptr)
+{
+  #ifdef DEBUG
```

Comment:
Your `ASSERT` definition must contain checking for `DEBUG`. So remove this preprocessor condition and move it to ASSERT module